src/api_server/src/lib.rs

```diff
@@ -82,14 +90,27 @@ impl ApiServer {
         let sessions = web::Data::new(SessionMap::new());
         let http_timeout = self.http_timeout;
 
+        let repository = self
+            .config
+            .repository_type
+            .to_repository(&self.config.repository_description)?;
+
         HttpServer::new(move || {
             App::new()
                 .wrap(middleware::Logger::default())
                 .app_data(web::Data::clone(&sessions))
                 .app_data(web::Data::clone(&attestation_service))
+                .app_data(web::Data::new(repository.clone()))
```

Comment:
As the ownership of `repository` is transferred to this function, we might delete the `clone` here to reduce the memory usage.

Reply:
If we don't use clone here:

``` 
error[E0507]: cannot move out of `repository`, a captured variable in an `Fn` closure
   --> src/api_server/src/lib.rs:103:42
    |
93  |         let repository = self
    |             ---------- captured outer variable
...
98  |         HttpServer::new(move || {
    |                         ------- captured by this `Fn` closure
...
103 |                 .app_data(web::Data::new(repository))
    |                                          ^^^^^^^^^^ move occurs because `repository` has type `Arc<dyn Repository + Send + Sync>`, which does not implement the `Copy` trait

For more information about this error, try `rustc --explain E0507`.
```

Reply:
Sorry, you're right. I made a mistake, :-)